### ISSUE TITLE
Fabric with ddp_notebook causes RuntimeError: Default process group has not been initialized

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/lightning#18320]
### Bug description

When running Fabric code with `strategy="ddp_notebook"`, it raises an error:

```python
RuntimeError: Default process group has not been initialized, please make sure to call init_process_group
```

while it works fine with regular `strategy="ddp"` in a script.

(via @XinzeWu here: https://github.com/rasbt/cvpr2023/issues/1)

### What version are you seeing the problem on?

v2.0

### How to reproduce the bug

```python
See https://github.com/rasbt/cvpr2023/issues/1
```

### Error messages and logs

```
RuntimeError: Default process group has not been initialized, please make sure to call init_process_group
```

cc @carmocca @justusschock @awaelchli

**Comment by XinzeWu** [comment]
and ddp is not fitable for jupyter-notebook
with bug:
RuntimeError: `Fabric(strategy='ddp')` is not compatible with an interactive environment. Run your code as a script, or choose one of the compatible strategies: `Fabric(strategy='dp'|'ddp_notebook')`. In case you are spawning processes yourself, make sure to include the Fabric creation inside the worker function.

**Comment by rasbt** [comment]
> RuntimeError: Fabric(strategy='ddp') is not compatible with an interactive environment. 

That's a good point but expected since interactive environments in Python have some limitations and things like `torch.multiprocessing.spawn` (which ddp uses) wouldn't work there. So `"ddp_notebook"` is basically a workaround for that.

**Comment by awaelchli** [comment]
Hi @rasbt and @XinzeWu 

Thanks for opening the issue. There are two things I noticed here:
1. The code in the linked issue does not follow the requirements for running in a notebook. The training function needs to be passed as input to `fabric.launch()`. See the guide here: https://lightning.ai/docs/fabric/stable/fundamentals/notebooks.html#fabric-in-notebooks
2. The code uses Lightning 2.0.x if I understand correctly. In 2.1, we've added a bit more validation with useful error messages should the usage be incorrect, see #17423 and #17570. These measures should prevent you running into this again and give the user better instructions without requiring to necessarily go to the docs in the first place. 

Let me know if that helps.
